test/test_js_instrument.py

```diff
@@ -1,129 +1,286 @@
+            symbol_prefix='',
+            doc_url=self.TOP_URL,
+            top_url=self.TOP_URL,
+            expected_method_calls=self.METHOD_CALLS,
+            expected_gets_and_sets=self.GETS_AND_SETS,
+        )
+
+
+class TestJSInstrumentExistingWindowProperty(OpenWPMJSTest):
+
+    GETS_AND_SETS = {
+        ("window.partiallyExisting", "get", "{\"existingProp\":\"foo\"}"),
+        ("window.partiallyExisting", "get", "{\"existingProp\":\"foo\"}"),
+        ("window.partiallyExisting", "get", "{\"existingProp\":\"foo\"}"),
+        ("window.partiallyExisting", "get", "{\"existingProp\":\"foo\"}"),
+        ("window.partiallyExisting.existingProp", "get", "foo"),
+        ("window.partiallyExisting", "get", "{\"existingProp\":\"foo\"}"),
+        ("window.partiallyExisting.existingProp", "set", "blah1"),
+        ("window.partiallyExisting", "get", "{\"existingProp\":\"blah1\"}"),
+        ("window.partiallyExisting.existingProp", "get", "blah1"),
+        ("window.partiallyExisting", "get", "{\"existingProp\":\"blah1\"}"),
+        ("window.partiallyExisting.nonExistingProp1", "get", "undefined"),
+        ("window.partiallyExisting", "get", "{\"existingProp\":\"blah1\"}"),
+        ("window.partiallyExisting.nonExistingProp1", "set", "blah1"),
+        ("window.partiallyExisting", "get", "{\"existingProp\":\"blah1\"}"),
+        ("window.partiallyExisting.nonExistingProp1", "get", "blah1"),
+        ("window.partiallyExisting", "get",
+         "{\"existingProp\":\"blah1\"}"),  # Note 1
+        ("window.partiallyExisting.nonExistingMethod1",
+         "get", "undefined"),  # Note 2
+    }
+
+    # Note 1: nonExistingProp1 is not enumerable even after being set
+    # Note 2: nonExistingMethod1 shows up as a get rather than call
+
+    METHOD_CALLS = set()  # Note 2
+
+    TEST_PAGE = "instrument_existing_window_property.html"
+    TOP_URL = (
+        u"%s/js_instrument/%s" % (util.BASE_TEST_URL, TEST_PAGE)
+    )
+
+    def test_instrument_object(self):
+        """ Ensure instrumentObject logs all property gets, sets, and calls """
+        db = self.visit('/js_instrument/%s' % self.TEST_PAGE)
+        self._check_calls(
+            db=db,
+            symbol_prefix='',
+            doc_url=self.TOP_URL,
+            top_url=self.TOP_URL,
+            expected_method_calls=self.METHOD_CALLS,
+            expected_gets_and_sets=self.GETS_AND_SETS,
+        )
+
+
+class TestJSInstrumentExistingWindowPropertyByPython(TestJSInstrumentExistingWindowProperty):  # noqa
+    # This test is a repeat of above test, but as an integration test
+    # to test correct setting of instrumentaiton via browser params.
 
     def get_config(self, data_dir=""):
-        manager_params, browser_params = self.get_test_config(data_dir)
-        browser_params[0]['js_instrument'] = True
-        manager_params['testing'] = True
+        manager_params, browser_params = super().get_config(data_dir)
+        browser_params[0]['js_insrument_modules'] = [
```

Comment:
It is running (https://travis-ci.org/github/mozilla/OpenWPM/jobs/703203803#L1153)

It is picking up the mis-named setting: https://travis-ci.org/github/mozilla/OpenWPM/jobs/703203803#L1205

And it's passing. :s

And it's passing because it's using the same test page which has the JS setting in! Great catch. This needs fixing. Thanks.

Reply:
Fixed in 9559ec9.

This actually was a bigger fix because you can't instrument properties that don't exist. So the other tests work because they build the object, then call the window.instrumentObject function. However, pyside the instrumentation loads before page load (which we want, so we don't miss anything) but it means that we can't instrument arbitrary objects, so I instrument real objects. Which had the nice side effect of me ending up getting a better understanding of how the instrumentation works, so I added a little extra to the README.